src/main/groovy/life/qbic/db/MariaDBManager.groovy

```diff
@@ -466,14 +488,22 @@ class MariaDBManager implements IQueryService {
     final String query = "SELECT * from samples WHERE UPPER(id) = UPPER('${sampleId}');"
     try {
       List<GroovyRowResult> results = sql.rows(query)
-      if( results.size() > 0 ) {
+      if (results.size() > 0) {
         GroovyRowResult rs = results.get(0)
         int locationID = rs.get("current_location_ID")
         setStatus(sampleId, locationID, status, sql)
+      } else {
+        throw new NotFoundException("Sample $sampleId could not be found in the database.")
       }
+    } catch (NotFoundException notFoundException) {
+      sql.close()
+      throw notFoundException
     } catch (SQLException e) {
+      log.error("Could not update sample status for $sampleId: $e.message")
+      log.debug("Could not update sample status for $sampleId: $e.message", e)
```

Comment:
are both needed?

Reply:
one logs the whole stack trace (into debug). the other call only the message. So to not clutter the log I would say yes :)

Reply:
agreed